From a96b9809a13453af2bab8408d482b20404be6243 Mon Sep 17 00:00:00 2001
From: Kazunori Kobayashi <kkobayas@igel.co.jp>
Date: Tue, 9 Apr 2013 15:40:27 +0900
Subject: [PATCH 1/2] ext/dfbvideosink: fix a destination address of CbCr
 plane

viosurface[DST].pc was specified as an incorrect address that
indicates out of the CbCr plane in the framebuffer.
This patch appropriately sets the CbCr plane address.
---
 ext/directfb/dfbvideosink.c |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/ext/directfb/dfbvideosink.c b/ext/directfb/dfbvideosink.c
index 39aa4cc..c2a4149 100644
--- a/ext/directfb/dfbvideosink.c
+++ b/ext/directfb/dfbvideosink.c
@@ -2258,7 +2258,8 @@ gst_dfbvideosink_shvio_stretchblit (GstDfbVideoSink * dfbvideosink,
     if (is_ycbcr (viosurface[DST].format)
         && viosurface[SRC].format != REN_UYVY)
       viosurface[DST].pc = (void *) (
-          (gulong) viosurface[DST].py + viosurface[DST].pitch * dst->h);
+          (gulong) viosurface[DST].py + viosurface[DST].pitch *
+          (dfbvideosink->out_height - dst->y / 2));
     else
       viosurface[DST].pc = 0;
 #if defined(HAVE_SHMERAM)
-- 
1.7.5.4

